x/concentrated-liquidity/internal/math/math.go

```diff
@@ -12,9 +14,15 @@ func Liquidity0(amount sdk.Int, sqrtPriceA, sqrtPriceB sdk.Dec) sdk.Dec {
 	if sqrtPriceA.GT(sqrtPriceB) {
 		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
 	}
-	product := sqrtPriceA.Mul(sqrtPriceB)
-	diff := sqrtPriceB.Sub(sqrtPriceA)
-	return amount.ToDec().Mul(product).Quo(diff)
+
+	// transform sqrtPriceA and sqrtPriceB to BigDec using osmomath for more final precision
+	sqrtPriceAOsmomath := osmomath.MustNewDecFromStr(sqrtPriceA.String())
+	sqrtPriceBOsmomath := osmomath.MustNewDecFromStr(sqrtPriceB.String())
+	product := sqrtPriceAOsmomath.Mul(sqrtPriceBOsmomath)
+	diff := sqrtPriceBOsmomath.Sub(sqrtPriceAOsmomath)
+
+	// conduct the liquidity calculation, then transform back to normal sdk.Dec
+	return osmomath.MustNewDecFromStr(amount.String()).Mul(product).Quo(diff).SDKDec()
```

Comment:
We should not construct this from a string. There should be helpers to perform the conversion `Int` -> `Dec` -> `BigDec`

Reply:
Offline, agreed to revert liquidity0 and liquidity1 back to normal sdk.Decs due to minimal improvement in precision